### ISSUE TITLE
zsh-autosuggestions style autocomplete

### ISSUE THREAD

**Issue by joooooooooooooooooooooooooooooooooooosh** [insou22/mipsy#121]
Could we give C-f the same behaviour as the right arrow key?
If the cursor is at the end of the line the right arrow key autocompletes the suggested command, otherwise it just moves the cursor forward.
Currently C-f only moves the cursor forward, even if the end of the line has already been reached.

(this is probably handled inside the rustyline crate so idk how easily this can be done)

**Comment by joooooooooooooooooooooooooooooooooooosh** [comment]
Looking into it further, binding this would be a very simple change within the rustyline crate itself, but as of now the API support for conditional/overloaded keybindings is basically nonexistent. Not possible to neatly implement within mipsy unfortunately

**Comment by insou22** [comment]
Looks like this still doesn't work after the rustyline upgrade, is that as expected @joooooooooooooooooooooooooooooooooooosh ?

**Comment by joooooooooooooooooooooooooooooooooooosh** [comment]
Yeah it still isn't implemented, upgrading rustyline was just a side effect of me trying to get it working in the first place.
However this will make it easy to fix if my rustyline PR gets added or I decide to bother making use of ConditionalEventHandlers which are a newer rustyline feature.
